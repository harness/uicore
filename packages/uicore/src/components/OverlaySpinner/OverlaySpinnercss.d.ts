/*
 * Copyright (c) Harness Inc.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly overlayLoader: string
  readonly overlay: string
}
export default styles
