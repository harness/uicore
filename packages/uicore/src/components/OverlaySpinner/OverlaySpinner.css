/*
 * Copyright (c) Harness Inc.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

.overlaySpinner {
  position: relative;
}

.overlay {
  position: absolute;
  background: rgb(0, 0, 0);
  background: rgba(0, 0, 0, 0.2);
  -webkit-transition: 0.5s ease;
  transition: 0.5s ease;
  height: 100%;
  width: 100%;
  z-index: 15;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
}
