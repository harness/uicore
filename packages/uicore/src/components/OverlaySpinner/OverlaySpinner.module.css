/*
 * Copyright 2020 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.overlaySpinner {
  position: relative;
}

.overlay {
  position: absolute;
  background: rgb(0, 0, 0);
  background: rgba(0, 0, 0, 0.2);
  -webkit-transition: 0.5s ease;
  transition: 0.5s ease;
  height: 100%;
  width: 100%;
  z-index: 15;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
}
