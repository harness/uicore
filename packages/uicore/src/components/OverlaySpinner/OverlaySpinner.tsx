/*
 * Copyright 2020 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import css from './OverlaySpinner.css'
import { ISpinnerProps, Spinner } from '@blueprintjs/core'

export interface OverlaySpinnerProps extends ISpinnerProps {
  show: boolean
  children: React.ReactNode
}

export const OverlaySpinner: React.FC<OverlaySpinnerProps> = props => {
  const { show, children, ...rest } = props
  return (
    <div className={css.overlaySpinner}>
      {children}
      {show ? (
        <div className={css.overlay}>
          <Spinner {...rest} />
        </div>
      ) : null}
    </div>
  )
}
