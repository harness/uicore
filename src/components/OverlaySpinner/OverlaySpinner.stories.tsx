import React from 'react'
import type { Meta, Story } from '@storybook/react'
import { Title, Subtitle, ArgsTable, Stories, PRIMARY_STORY, Primary, Description } from '@storybook/addon-docs/blocks'
import { OverlaySpinner, OverlaySpinnerProps, Layout, Text } from '../..'
import { omit } from 'lodash-es'

export default {
  title: 'Components / OverlaySpinner',

  component: OverlaySpinner,
  parameters: {
    docs: {
      source: {
        type: 'code'
      },

      page: function PageDescription() {
        return (
          <>
            <Title>OverlaySpinner</Title>
            <Subtitle>
              <pre>
                <code>{`import {OverlaySpinner} from '@wings-software/uicore'`}</code>
              </pre>
            </Subtitle>
            <Description>{`Adds [Blueprint spinner](https://blueprintjs.com/docs/#core/components/spinner) to the overlay and blocks the user interaction
`}</Description>

            <Primary />
            <ArgsTable story={PRIMARY_STORY} />

            <Stories />
          </>
        )
      }
    }
  },
  decorators: [Story => <Story />]
} as Meta
export const Basic: Story<OverlaySpinnerProps> = args => {
  const { show = true } = args
  const argsCopy = omit(args, ['show'])
  return (
    <OverlaySpinner show={show} {...argsCopy}>
      <Layout.Vertical id="text-font-size">
        <Text>The quick brown fox jumps over the lazy dog</Text>

        <Text font="small">The quick brown fox jumps over the lazy dog</Text>
        <Text font="normal">The quick brown fox jumps over the lazy dog</Text>
        <Text font="medium">The quick brown fox jumps over the lazy dog</Text>
        <Text font="large">The quick brown fox jumps over the lazy dog</Text>
      </Layout.Vertical>
    </OverlaySpinner>
  )
}
