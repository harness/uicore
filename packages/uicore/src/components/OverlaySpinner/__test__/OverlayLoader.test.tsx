/*
 * Copyright 2020 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, queryByAttribute } from '@testing-library/react'
import { OverlaySpinner } from '../OverlaySpinner'

jest.useFakeTimers()

const DummyComponent = () => <div>DummyComponent</div>

describe('OverlayLoader Test', () => {
  test('', () => {
    const { rerender } = render(
      <OverlaySpinner show={false}>
        <DummyComponent />
      </OverlaySpinner>
    )
    expect(document.body).toMatchSnapshot()
    rerender(
      <OverlaySpinner show={true}>
        <DummyComponent />
      </OverlaySpinner>
    )
    expect(queryByAttribute('class', document.body, 'bp3-spinner')).not.toBeNull()
    expect(document.body).toMatchSnapshot()
  })
})
