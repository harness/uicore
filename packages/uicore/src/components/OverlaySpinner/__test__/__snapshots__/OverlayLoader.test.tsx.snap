// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverlayLoader Test  1`] = `
<body>
  <div>
    <div
      class="overlaySpinner"
    >
      <div>
        DummyComponent
      </div>
    </div>
  </div>
</body>
`;

exports[`OverlayLoader Test  2`] = `
<body>
  <div>
    <div
      class="overlaySpinner"
    >
      <div>
        DummyComponent
      </div>
      <div
        class="overlay"
      >
        <div
          class="bp3-spinner"
        >
          <div
            class="bp3-spinner-animation"
          >
            <svg
              height="50"
              stroke-width="8.00"
              viewBox="1.00 1.00 98.00 98.00"
              width="50"
            >
              <path
                class="bp3-spinner-track"
                d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
              />
              <path
                class="bp3-spinner-head"
                d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                pathLength="280"
                stroke-dasharray="280 280"
                stroke-dashoffset="210"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
